packages/react/src/Text/Text.tsx

```diff
@@ -58,14 +58,21 @@ const StyledText = styled.span<StyledTextProps>`
   ${sx};
 `
 
+const includesSystemProps = (props: StyledTextProps) => {
+  return (
+    props.sx ||
+    Object.keys(props).some(prop => Object.keys(TYPOGRAPHY).includes(prop) || Object.keys(COMMON).includes(prop))
+  )
+}
```

Comment:
One alternative for this could be to use a `Set` to keep track of the names so we don't have to recreate the `Object.keys` for `TYPOGRAPHY` and `COMMON` each type. I think the `Set` could also help for lookup types over `.includes` but definitely not sure 🤷 

```suggestion
const COMMON_PROP_NAMES = new Set(Object.keys(COMMON));
const TYPOGRAPHY_PROP_NAMES = new Set(Object.keys(TYPOGRAPHY));

const includesSystemProps = (props: StyledTextProps) => {
  if (props.sx) {
    return true;
  }
  
  return Object.keys(props).some((prop) => {
    return TYPOGRAPHY_PROP_NAMES.has(prop) ?? COMMON_PROP_NAMES.has(prop)
  });
}
```

Reply:
I like it 👍🏻